neo/SmartContract/Native/Tokens/NeoToken.cs

```diff
@@ -0,0 +1,422 @@
+using System.IO;
+using System.Linq;
+using System.Numerics;
+using VMArray = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Native.Tokens
+{
+    public class NeoToken : NativeContractBase
+    {
+        public const string ServiceName = "Neo.Native.Tokens.NEO";
+        public static readonly byte[] Script = CreateNativeScript(ServiceName);
+        public static readonly UInt160 ScriptHash = Script.ToScriptHash();
+        public static readonly string[] SupportedStandards = { "NEP-5", "NEP-10" };
+        public const string Name = "NEO";
+        public const string Symbol = "neo";
+        public const int Decimals = 0;
+        public static readonly BigInteger DecimalsFactor = BigInteger.Pow(10, Decimals);
+        public static readonly BigInteger TotalAmount = 100000000 * DecimalsFactor;
+
+        private const byte Prefix_Initialized = 11;
+        private const byte Prefix_Account = 20;
+        private const byte Prefix_Validator = 33;
+        private const byte Prefix_ValidatorsCount = 15;
+
+        internal static bool Main(ApplicationEngine engine)
+        {
+            if (!new UInt160(engine.CurrentContext.ScriptHash).Equals(ScriptHash))
+                return false;
+            string operation = engine.CurrentContext.EvaluationStack.Pop().GetString();
+            VMArray args = (VMArray)engine.CurrentContext.EvaluationStack.Pop();
+            StackItem result;
+            switch (operation)
+            {
+                case "supportedStandards":
+                    result = SupportedStandards.Select(p => (StackItem)p).ToList();
+                    break;
+                case "name":
+                    result = Name;
+                    break;
+                case "symbol":
+                    result = Symbol;
+                    break;
+                case "decimals":
+                    result = Decimals;
+                    break;
+                case "totalSupply":
+                    result = TotalAmount;
+                    break;
+                case "balanceOf":
+                    result = BalanceOf(engine, args[0].GetByteArray());
+                    break;
+                case "transfer":
+                    result = Transfer(engine, args[0].GetByteArray(), args[1].GetByteArray(), args[2].GetBigInteger());
+                    break;
+                case "initialize":
+                    result = Initialize(engine);
+                    break;
+                case "unclaimedGas":
+                    result = UnclaimedGas(engine, args[0].GetByteArray(), (uint)args[1].GetBigInteger());
+                    break;
+                case "registerValidator":
+                    result = RegisterValidator(engine, args[0].GetByteArray());
+                    break;
+                case "vote":
+                    result = Vote(engine, args[0].GetByteArray(), ((VMArray)args[1]).Select(p => p.GetByteArray().AsSerializable<ECPoint>()).ToArray());
+                    break;
```

Comment:
Perhaps `vote` and `registerValidator` should be part of another contract, a voting contract, which controls de governance of the network. So, in that contract, they could require Native Neo assets to allow voting, and control this logic from there. This better separates the role of Neo from the network governance. Opinions?

Reply:
A general voting contract is under development, and after it has been developed, we can reevaluate whether it is necessary to move the function of the consensus node election to that contract. Now let's implement it according to the current design.